stackit/internal/services/iaas/network/resource.go

```diff
@@ -141,20 +165,78 @@ func (r *networkResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 				},
 			},
 			"nameservers": schema.ListAttribute{
-				Description: "The nameservers of the network.",
+				Description:        "This field is deprecated and will be removed after April 28th 2025, use `ipv4_nameservers` to configure the nameservers for the IPv4 networks.",
+				DeprecationMessage: "Use `ipv4_nameservers` to configure the nameservers for the IPv4 networks.",
+				Optional:           true,
+				Computed:           true,
+				ElementType:        types.StringType,
+			},
+			"ipv4_gateway": schema.StringAttribute{
+				Description: "The IPv4 gateway of a network. If not specified, the first IP of the network will be assigned as the gateway.",
 				Optional:    true,
 				Computed:    true,
-				PlanModifiers: []planmodifier.List{
-					listplanmodifier.UseStateForUnknown(),
+				Validators: []validator.String{
+					validate.IP(),
 				},
+			},
+			"ipv4_nameservers": schema.ListAttribute{
+				Description: "The IPv4 nameservers of the network.",
+				Optional:    true,
+				Computed:    true,
 				ElementType: types.StringType,
 			},
+			"ipv4_prefix": schema.StringAttribute{
+				Description: "The IPv4 prefix of the network (CIDR).",
+				Optional:    true,
+				Validators: []validator.String{
+					validate.CIDR(),
+				},
+			},
 			"ipv4_prefix_length": schema.Int64Attribute{
 				Description: "The IPv4 prefix length of the network.",
 				Optional:    true,
 			},
 			"prefixes": schema.ListAttribute{
-				Description: "The prefixes of the network.",
+				Description:        "This field is deprecated and will be removed after April 28th 2025, use `ipv4_prefixes` to read the prefixes of the IPv4 networks.",
+				DeprecationMessage: "Use `ipv4_prefixes` to read the prefixes of the IPv4 networks.",
```

Comment:
```suggestion
			"prefixes": schema.ListAttribute{
				Description:        "The prefixes of the network. This field is deprecated and will be removed after April 28th 2025, use `ipv4_prefixes` to read the IPv4 prefixes of the network.",
				DeprecationMessage: "Use `ipv4_prefixes` to read the IPv4 prefixes of the network.",
```

Reply:
Remember to also update the descriptions in the datasource to match these

Reply:
updated descriptions in resource and datasource :)